### PR TITLE
CHERI-RISC-V: CSetEqualExact instruction and builtin

### PR DESCRIPTION
**Pull request by nwf**
Just cribbing from CTestSubset.  Apologies if I've missed something.

### CODE REVIEW

File: llvm/test/CodeGen/CHERI-Generic/RISCV32/intrinsics.ll [hunk 1]
```diff
@@ -484,3 +484,20 @@ define i32 @subset_test(i8 addrspace(200)* %cap1, i8 addrspace(200)* %cap2) noun
   %subset.zext = zext i1 %subset to i32
   ret i32 %subset.zext
 }
+
```
Comment (jrtc27) [inline llvm/test/CodeGen/CHERI-Generic/RISCV32/intrinsics.ll:487]: Second line of this file says:
```ll
; DO NOT EDIT -- This file was generated from test/CodeGen/CHERI-Generic/Inputs/intrinsics.ll
```

File: llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td [hunk 2]
```diff
@@ -339,6 +341,7 @@ def PseudoCapTAILIndirect : Pseudo<(outs), (ins GPCRTC:$rs1),
 
 let Predicates = [HasCheri] in {
 def CTestSubset : Cheri_rr<0x20, "ctestsubset", GPR, GPCR>;
+def CSetEqualExact : Cheri_rr<0x21, "csetequalexact", GPR, GPCR>;
```
Comment (jrtc27) [inline llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td:344]: CSetEqualExact should, IMO, be treated like CRRL/CRAM, in that its CSEQX short form is the preferred alias and the internal name of the instruction.

**Comment by nwf** [comment]
I do not know how to do the requisite lowering on MIPS, and I really do not know how to do the pattern recognition to produce CNEXEQ, though I took an initial stab at putting some of the code in place based on pattern matching against CTestSubset.

**Comment by jrtc27** [comment]
@nwf @arichardson I believe this is now complete.

**Comment by nwf** [comment]
Well that was eye-opening to read.  Thank you.

**Comment by jrtc27** [comment]
In hindsight I should probably have made the MIPS-specific SD nodes use i32 rather than i64 given i32 is MIPS's condition code type even on MIPS64, as the CEXEQ vs CNEXEQ32 thing looks rather strange... Then again, maybe I tried it and it didn't work, I don't remember, I've tried to block out all my memory of getting bool intrinsics to work without highly-inefficient code generation.

**Review by arichardson — APPROVED** [review]

**Pull request merged**